src/cryptsetup/cryptsetup-generator.c

```diff
@@ -37,9 +38,66 @@ static Hashmap *arg_disks = NULL;
 static char *arg_default_options = NULL;
 static char *arg_default_keyfile = NULL;
 
+static int generate_keydev_mount(const char *name, const char *keydev, char **unit, char **mount) {
+        _cleanup_free_ char *u = NULL, *what = NULL, *where = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        assert(name);
+        assert(keydev);
+        assert(unit);
+        assert(mount);
+
+        r = mkdir_p("/run/systemd/cryptsetup", 0755);
+        if (r < 0)
+                return r;
+
+        r = chmod("/run/systemd/cryptsetup", 0700);
```

Comment:
In practice this probably doesn't matter, but this introduces a race, where the the directory is created with wrong permissions, and then the permissions are updated. In principle somebody could use inotify to watch the directory creation and open the directory before the permissions are changed. Hence, let's just make the permissions correct from the start, so maybe use `mkdir_parents` and then create the final directory with the right permissions, without a separate `chmod`.